x-pack/plugins/serverless_search/public/navigation_tree.ts

```diff
@@ -5,181 +5,188 @@
  * 2.0.
  */
 
-import type { NavigationTreeDefinition } from '@kbn/core-chrome-browser';
+import type { AppDeepLinkId, NavigationTreeDefinition } from '@kbn/core-chrome-browser';
+import type { ApplicationStart } from '@kbn/core-application-browser';
 import { i18n } from '@kbn/i18n';
 import { CONNECTORS_LABEL } from '../common/i18n_string';
 
-export const navigationTree = (): NavigationTreeDefinition => ({
```

Comment:
The diffing algorithm makes it very hard to read what actually changed here. Could you summarize the intent here?

Reply:
The main change is in line 85: we add the `searchPlayground` entry conditionally, aka only if the `searchPlayground` application is registered (see `isAvailable`).

This makes use of the new Core API to check whether an app is registered, without having to explicitly depend on its plugin.

This loosens the coupling between plugins, so that there aren't any technical constraints to choose the right categorisation.